imod/msw/sprinkling.py

```diff
@@ -55,23 +55,29 @@ def __init__(
         self,
         max_abstraction_groundwater: xr.DataArray,
         max_abstraction_surfacewater: xr.DataArray,
-        well: WellDisStructured,
+        well: Mf6Wel,
     ):
         super().__init__()
         self.dataset["max_abstraction_groundwater_m3_d"] = max_abstraction_groundwater
         self.dataset["max_abstraction_surfacewater_m3_d"] = max_abstraction_surfacewater
+        if not isinstance(well, Mf6Wel):
+            raise TypeError(rf"well not of type 'Mf6Wel', got '{type(well)}'")
         self.well = well
 
         self._pkgcheck()
 
     def _render(self, file, index, svat):
-        well_row = self.well["row"] - 1
-        well_column = self.well["column"] - 1
-        well_layer = self.well["layer"]
+        cellid = self.well["cellid"]
+        if len(cellid.coords["nmax_cellid"]) != 3:
+            raise TypeError("Coupling to unstructured grids is not supported.")
+
+        well_layer = cellid.sel(nmax_cellid="layer").data
```

Comment:
What is `nmac_cellid`? I cant figure it our by just looking at this file
Could you rename it to something more describing?

Reply:
I based this name on an example I found in the UGRID conventions. There it is named ``nMaxMesh2_face_nodes``, meaning max amount of nodes per face. The "max" is relevant in Ugrids, as meshes can be flexible so the amount of nodes per face can vary in the grid. 

In our case this is not happening, so the "max" is wrong here. I'll rename to ``ncellid``, as that is a better name for this dimension.

Reply:
+1 for `ndim_cellid`

Reply:
After a brief discussion with @Manangka, I renamed it to ``dim_cellid``.